railties/lib/rails/application/credentials_config.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Rails
+  class Application
+    class CredentialsConfig # :nodoc:
```

Comment:
I think adding this class may be overkill.  Something like #47149 would suffice.

Reply:
Yes. It would. I'm writing up the description right now but wanted to make sure the build is green (which it is).

I had this roughed in first, but it felt like a pretty niche feature. I'm facepalming now that this bug has been found and there's an open PR for it.